dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1788,6 +1788,7 @@ private Map<String, Object> checkParams(User loginUser,
         return result;
     }
 
+    @Transactional(rollbackFor = RuntimeException.class)
```

Comment:
`this.doBatchOperateProcessDefinition`  will make the `@Transactional` invalidated, you can hold a reference of ProcessDefinitionServiceImpl and use the reference to call doBatchOperateProcessDefinition.